<p>You've configured <code>wait()</code> to return upon the sooner of the timer finishing or the user cancelling the loop. It sounds like you want to cancel the <code>count_timer</code> in the case that it <em>wasn't</em> the future that caused the <code>wait</code> call to return. You can learn this simply by asking which of the two futures is done.</p>

<pre><code>count_timer = asyncio.sleep(COUNT_TIMEOUT)
await asyncio.wait([count_timer, stop_future], return_when=FIRST_COMPLETED)
if not count_timer.done():
    count_timer.cancel()
</code></pre>
